AppActionVisualizer/app/src/main/java/com/example/appactionvisualizer/ui/activity/dashboard/DeepLinkListActivity.java

```diff
@@ -1,69 +1,413 @@
+    matched.put(actionName, entry.getValue());
+    return matched;
+  }
+
+  /**
+   * @param allActions The actions pool to be selected, key is the action name.
+   * @param inputWords all the words of user input.
+   * @return the most likely action(s) among all actions.
+   */
+  private Map<String, List<AppFulfillment>> getActionFromWords(
+      Map<String, List<AppFulfillment>> allActions, String[] inputWords) {
+    // Has to be tree map since we wanna a ordered score from low to high.
+    TreeMap<Integer, List<String>> scoresMap = new TreeMap<>();
+    // Iterate over all actions to find the most likely one.
+    for (Map.Entry<String, List<AppFulfillment>> entry : allActions.entrySet()) {
+      String actionName = entry.getKey();
+      String[] tobeMatched = actionNameWords.get(actionName);
+      if (tobeMatched == null || tobeMatched.length == 0) {
+        continue;
+      }
+      // Compute score to get a priority of this action.
+      int score = StringUtils.matchScore(tobeMatched, inputWords);
+      if (scoresMap.get(score) == null) {
+        scoresMap.put(score, new ArrayList<String>());
+      }
+      scoresMap.get(score).add(actionName);
+    }
+    List<String> matchedActions = scoresMap.lastEntry().getValue();
+    // Keep the map order as insertion order
+    Map<String, List<AppFulfillment>> matched = new HashMap<>();
+    for (String actionName : matchedActions) {
+      matched.put(actionName, allActions.get(actionName));
+    }
+    return matched;
+  }
+
+  /**
+   * @param appAction an app action to be parsed
+   * @return Construct an data map from single AppAction.
+   */
+  private Map<String, List<AppFulfillment>> parseActionsFromApp(AppAction appAction) {
+    Map<String, List<AppFulfillment>> resultMap = new TreeMap<>(comparator);
+    for (Action action : appAction.getActionsList()) {
+      String intentName = action.getIntentName();
+      resultMap.put(intentName, new ArrayList<AppFulfillment>());
+      for (FulfillmentOption fulfillmentOption : action.getFulfillmentOptionList()) {
+        resultMap.get(intentName).add(new AppFulfillment(appAction, action, fulfillmentOption));
+      }
+    }
+    return resultMap;
+  }
+
+  /**
+   * @param displayMap the content of expandable list view.
+   * Update expandable list view by passing some data.
+   * Parse null to indicate default data.
+   */
+  private void updateView(Map<String, List<AppFulfillment>> displayMap) {
+    // If no recommended data, display the default data.
+    boolean unfold = true;
+    if (displayMap == null)  {
+      displayMap = defaultMap;
+      unfold = false;
+    }
+    actionNames.clear();
+    intentMap.clear();
+    actionNames.addAll(displayMap.keySet());
+    intentMap.putAll(displayMap);
+    adapter.notifyDataSetChanged();
+    // Only unfold the first group when having recommended data
+    if(unfold) {
+      expandableListView.expandGroup(0);
+    }else {
```

Comment:
space after "if" and before "else"

Reply:
I think I forgot to use the format plugin after some commits:( Fixed.